conformance/tests/callables_annotation.py

```diff
@@ -129,14 +129,34 @@ def __call__(self, a: float, /, b: int, *, k: str, m: str) -> None:
         pass
 
 
-def func(p1: Proto1, p2: Proto2, p3: Proto3, p7: Proto7):
-    assert_type(p1, Callable[..., None])  # OK
-    assert_type(p1, Proto5[Any])  # E
-    assert_type(p2, Callable[Concatenate[int, ...], None])  # OK
-    assert_type(p3, Callable[..., None])  # E
-    assert_type(p3, Proto4[...])  # OK
-
-    f1: Proto6 = p7  # OK
+class Proto8(Protocol):
+    def __call__(self) -> None: ...
+
+
+def func5(
+    p1: Proto1,
+    p2: Proto2,
+    p3: Proto3,
+    p4: Proto4[...],
+    p5: Proto5[Any],
+    p7: Proto7,
+    p8: Proto8,
+    c1: Callable[..., None],
+    c2: Callable[Concatenate[int, ...], None],
+):
+    ok1: Callable[..., None] = p1  # OK
+    ok2: Proto1 = c1  # OK
+    ok3: Callable[..., None] = p5  # OK
+    ok4: Proto5[Any] = c1  # OK
+    ok5: Callable[Concatenate[int, ...], None] = p2  # OK
+    ok6: Proto2 = c2  # OK
+    ok7: Callable[..., None] = p3  # OK
+    ok8: Proto3 = c1  # OK
+    ok9: Proto4[...] = p3  # OK
+    ok10: Proto3 = p4  # OK
+    ok11: Proto6 = p7  # OK
+
+    err1: Proto5[Any] = p8  # E
```

Comment:
@erictraut Can you explain why this has to be an error? I don't understand why Proto5[Any] cannot be assigned by p8 if ok11 can be assigned by p7. Mypy works the same way as Zuban here. I can see how Proto6 and Proto7 should be assignable when reading the spec. However I wouldn't even know how to implement that in Mypy or Zuban. Can you let me know what your thought processes were here?